File: content-service/src/main/java/dev/animedia/contentservice/comicchapter/core/ComicChapter.java
package dev.animedia.contentservice.comicchapter.core;

import dev.animedia.contentservice.content.core.Content;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.UUID;

@Getter
@Setter
@Entity
@Table(
    name = "comic_chapter",
    check = {
        @CheckConstraint(
            name = "comic_chapter_pages_positive",
            constraint = "pages > 0"
        ),
        @CheckConstraint(
            name = "comic_chapter_episode_positive",
            constraint = "episode > 0"
        )
    }
)
public class ComicChapter {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID uuid;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "content_id")
    private Content content;

    @Column(nullable = false)
    private Integer pages;

    @Column(nullable = false)
    private Integer episode;

    @Column
    private boolean active = true;

    @Column
    private LocalDate releaseDate;
}

File: services/content-type-service/internal/domain/comic/comic_chapter.go
package comic

import "time"

type Chapter struct {
	ID          string
	ContentID   string
	Pages       int32
	Episode     int32
	Active      bool
	ReleaseDate time.Time
}

type ChapterImage struct {
	ID        string
	ChapterID string
	Page      int32
	ImageID   string
}

type ChapterTranslation struct {
	ID          string
	ChapterID   string
	LanguageID  string
	Name        string
	Description string
}
